src/lib/client.js

```diff
@@ -22,13 +23,18 @@ function initPreactIslandElement() {
 				var p = e.previousSibling;
 				while (p != s) {
 					if (!p || p == s) break;
-
 					e.parentNode.removeChild(p);
 					p = e.previousSibling;
 				}
-				while (d.firstChild) e.parentNode.insertBefore(d.firstChild, e);
 
-				d.parentNode.removeChild(d);
+				requestAnimationFrame(() => {
```

Comment:
This fixes a bug where the app server-side renders the whole `document`, we have to wait for the `d.children` to actually be populated which is an `afterFrame`